omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js

```diff
@@ -209,44 +138,20 @@ export default class ObsAcrossEncountersController {
    * @param encounters
    */
   updateWithConceptShortNames(encounters) {
-    if (encounters && encounters.length > 0) {
-      angular.forEach(encounters, encounter => {
-        let encounterObs = encounter.obs;
-        for (const prop in encounterObs) {
-          let obs = encounterObs[prop];
-          let conceptUuid = null;
-          // if the obs value is coded then we try to look for the short names of the coded answers
-          if (angular.isDefined(obs.value.concept) ) {
-            // some obs.value nodes contain the concept property that points to the concept
-            conceptUuid = obs.value.concept.uuid;
-          } else if (angular.isDefined(obs.value.uuid)) {
-            // and other coded obs(e.g. Drug Frequency) do not contain a child concept property
-            conceptUuid = obs.value.uuid;
-          }
-          if (conceptUuid !== null && conceptUuid.length > 0) {
-            this.openmrsRest.get("concept/" + conceptUuid, {
-              v: this.CONCEPT_CUSTOM_REP
-            }).then((concept) => {
-              let shortDisplay = this.getConceptWithShortName(concept);
-              obs.value.display = shortDisplay.display;
-            }, function (err) {
-              console.log(`failed to retrieve concept ${conceptUuid}, ${err}`);
-            });
-          }
+    for (let encounter of encounters) {
+      for (let obs of encounter.obs) {
+        // if the obs value is coded then we try to look for the short names of the coded answers
+        // some obs.value nodes contain the concept property that points to the concept
+        // and other coded obs(e.g. Drug Frequency) do not contain a child concept property
```

Comment:
The distinction here is whether the obs is valueCoded (so the value is a concept) or valueDrug (so the value is a drug), so whether obs.value = Concept or obs.value = Drug.

Drug objects have a concept property, so therefore if obs.value = Drug, then obs.value.concept.uuid is the concept associated with that drug.  So what I believe the chunk of code below is doing is saying "if this is a valueCoded, set the obs.value.display to the shortname of the concept, and if this is a valueDrug, set the obs.value.display to the shortname of the concept associated with the drug (when by default I believe it is the name of the drug)"

Reply:
Yyyyyes. This seems like what we want. Right?

Reply:
Yeah, I just thought the comment was a confusing and since what is happening here is confusing I wanted to make srue the comment was clear:  the difference is between obs that are "valueCoded" (the value is an Obs) and "valueDrug" (the value is a Drug).    In the 95% use care for the widget, will likely just be displaying "valueCoded"... this code is just to handle the 5% use case (which happens to be our use case here) where the value is is a valueDrug.  The comment makes it sound arbitrary ("some obs.value nodes contain the concept... and other coded obs ... do not").  I think there was confusion when this code was originally written why it needed to be that way, and I wanted to clear that up.